src/OpenTelemetry.Extensions/Internal/BaggageLogRecordProcessor.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.Logs;
+
+namespace OpenTelemetry.Extensions.Internal;
+
+internal sealed class BaggageLogRecordProcessor : BaseProcessor<LogRecord>
+{
+    private readonly Predicate<string> baggageKeyPredicate;
+
+    public BaggageLogRecordProcessor(Predicate<string> baggageKeyPredicate)
+    {
+        this.baggageKeyPredicate = baggageKeyPredicate ?? throw new ArgumentNullException(nameof(baggageKeyPredicate));
+    }
+
+    public static Predicate<string> AllowAllBaggageKeys => (_) => true;
+
+    public override void OnEnd(LogRecord data)
+    {
+        if (data != null && Baggage.Current.Count > 0)
+        {
+            var attributes = new List<KeyValuePair<string, object?>>();
+
+            foreach (var entry in Baggage.Current)
+            {
+                if (this.baggageKeyPredicate(entry.Key))
+                {
+                    attributes.Add(new(entry.Key, entry.Value));
+                }
+            }
+
+            if (data.Attributes != null)
+            {
+                attributes.AddRange(data.Attributes);
```

Comment:
It might be slightly faster to pass the existing attributes when initializing collection? Could lower the chances of resizing the underlying storage.

```csharp
var attributes = new List<KeyValuePair<string, object?>>(data.Attributes);
```

Reply:
or allocate enough capacity in List at ctor time to hold existing attributes+baggage.count.